common-libs/vc-common/src/verifier/util.ts

```diff
@@ -27,57 +27,61 @@ type Validations<T> = { [k in keyof T]: Validator | Validator[] }
 
 export type ValidateFn<T> = (data: Unvalidated<T>) => Promise<Validatied<T>>
 
-export const genValidateFn = <T>(validations: Validations<T>): ValidateFn<T> => async (data) => {
-  const keys = Object.keys(validations)
-  const errors: ErrorConfig[] = []
-
-  for (const _fieldName of keys) {
-    const fieldName = _fieldName as keyof T
-    const validator: Validator | Validator[] = validations[fieldName]
-    const validators: Validator[] = validator instanceof Array ? validator : [validator]
-
-    for (const validator of validators) {
-      try {
-        const outcome = await validator(data[fieldName], data)
-
-        if (outcome !== true) {
+export const genValidateFn = <T>(validations: Validations<T>): ValidateFn<T> => {
+  return async (data) => {
+    const keys = Object.keys(validations)
+    const errors: ErrorConfig[] = []
+
+    for (const _fieldName of keys) {
+      const fieldName = _fieldName as keyof T
+      const validator: Validator | Validator[] = validations[fieldName]
+      const validators: Validator[] = validator instanceof Array ? validator : [validator]
+
+      for (const validator of validators) {
+        try {
+          const outcome = await validator(data[fieldName], data)
+
+          if (outcome !== true) {
+            errors.push({
+              kind: 'invalid_param',
+              message: `Invalid value for field "${fieldName}": ${outcome.message}`,
+            })
+          }
+        } catch (err) {
           errors.push({
-            kind: 'invalid_param',
-            message: `Invalid value for field "${fieldName}": ${outcome.message}`,
+            kind: 'validation_error',
```

Comment:
Did you check, we are not relying on this error kind anywhere?

Reply:
@dmfilipenko-affinidi , hide whitespace changes (this file was reformatted, and github does not display all the changes correctly).

Reply:
I'm extremely confused how this diff could have made it into this PR, I just ran `eslint --fix`...

I'll rerun this again and make sure there are no functional changes

Reply:
I don't think there were functional changes here, it's just github showing them weirdly. But double check, of course